### PR TITLE
Add Jest test for PdfUIPageNumInput

### PR DESCRIPTION
**Pull request by meckila-navapbc**
Connects #14754
### Description
Rewrite karma test with jest test for PdfUIPageNmInput component

### Acceptance Criteria
- [x] Code compiles correctly
- [x] Remove karma test
- [x] Add jest test for component 
- [x] Update user-event library to use specialChars

### Storybook Story
*For Frontend (Presentational) Components*
* [x] Add Storybook story

### CODE REVIEW

File: client/app/reader/PdfUIPageNumInput.jsx [hunk 1]
```diff
@@ -59,7 +59,7 @@ export class PdfUIPageNumInput extends React.PureComponent {
     return (
       <div style={{ display: 'inline-block' }}>
         <TextField
-          maxLength="4"
```
Comment (meckila-navapbc) [inline client/app/reader/PdfUIPageNumInput.jsx:62]: This was an incorrect prop type and should have been a number

File: client/test/app/reader/PdfUIPageNumInput.test.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import { fireEvent, render, screen } from '@testing-library/react';
```
Comment (charleyf) [inline client/test/app/reader/PdfUIPageNumInput.test.js:2]:
```suggestion
import {render, screen } from '@testing-library/react';
```

File: client/test/app/reader/PdfUIPageNumInput.test.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+  const props = {
+    jumpToPage,
+    numPages: 20,
+    docId: 42
+  };
+
+  const setupTestComponent = () => {
+    return render(<PdfUIPageNumInput {...props} />);
+  };
```
Comment (charleyf) [inline client/test/app/reader/PdfUIPageNumInput.test.js:17]: - In a several places you call: `setupTestComponent(props)` no need to pass 'props' to that function.
- I don't think these props are ever changed for these tests. The below pattern that JC uses in DocketSwitchEditTasksForm.test.js  would make it easy to override props if future tests need to do that.

```
  const defaults = {
    jumpToPage,
    numPages: 20,
    docId: 42
  };

  const setupTestComponent = (props = {}) => {
    return render(<PdfUIPageNumInput {...defaults} {...props} />);
  };

 // Example of overriding: setupTestComponent({ numPages: 45 });
```

File: client/test/app/reader/PdfUIPageNumInput.test.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  it('renders properly', () => {
+    const component = setupTestComponent(props);
+
+    expect(component).toMatchSnapshot();
+  });
+
+  it('shows current page number', () => {
+    setupTestComponent(props);
+
+    expect(screen.getByLabelText('Page').value).toEqual('1');
```
Comment (charleyf) [inline client/test/app/reader/PdfUIPageNumInput.test.js:32]: It took me a minute to figure out that this is the default value, that's why it's '1'. Maybe the test could be `shows default page number of 1`

**Review by charleyf — CHANGES_REQUESTED** [review]
Looks good with a few comments. TIL about 'specialChars' I will certainly be using that in future tests.

**Comment by charleyf** [comment]
Oh! The karma test file can also be removed: client/test/karma/components/PdfUIPageNumInput-test.js

File: client/test/app/reader/PdfUIPageNumInput.test.js (hunk 3 above)
Comment (meckila-navapbc) [reply client/test/app/reader/PdfUIPageNumInput.test.js:17]: Ah this is leftover from when I though I needed to override props

File: client/package.json [hunk 5]
```diff
@@ -144,7 +144,7 @@
     "@testing-library/jest-dom": "^5.11.3",
     "@testing-library/react": "^10.4.9",
     "@testing-library/react-hooks": "^3.4.1",
-    "@testing-library/user-event": "^12.1.1",
+    "@testing-library/user-event": "^12.8.3",
```
Comment (meckila-navapbc) [inline client/package.json:147]: I bumped the version because  12.1.1 didn't include fix that exports `specialChars`

**Review by charleyf — APPROVED** [review]

**Pull request merged**